### ISSUE TITLE
Preferences panel: problematic `labels` (disconnected from their respective input fields)

### ISSUE THREAD

**Issue by phiw13** [textpattern/textpattern#1920]
1/ Missing ID under Advanced options - “Custom form template” and “file_download_header”

``` html
<label for="custom_form_types">Custom form template types&nbsp;<a rel="help" title="Help" role="button"[…] >Help</span></a></label> 

<label for="file_download_header">file_download_header</label>
```

--> The `for` attribute for both `<label />`s are referencing a non-existing ID – the relevant `<textarea />` only specifies a `name` attribute, no `id`.

2/ In other Preferences sub groups (Site and Mail)

Two fields with similar problem: `<label />` should *not* be used here (the radio buttons have a `<label />` on the “Yes/No” text), but it probably could use `aria-labeled-by` as suggested in an other issue)

``` html
<label for="trailing_slash">Add a trailing slash to URLs</label>

<label for="enhanced_email">Use enhanced mail features?</label>
```

**Comment by bloatware** [comment]
1) is easy to fix, 2) is more annoying.

**Comment by bloatware** [comment]
Yes, the second point should be fixed more properly. Currently, all pref input handler names (like `yesnoradio`) that do not require a common label are hard-coded. As soon as a (new) pref uses a custom handler that is not on the list (like `trailing_slash`), it gets labelled.

**Comment by Bloke** [comment]
@bloatware Any further work required here or can we close this?

**Comment by bloatware** [comment]
If @phiw13 is happy with it I'm happy too.

**Comment by phiw13** [comment]
Keeping this [comment](https://github.com/textpattern/textpattern/issues/1920#issuecomment-2104489332) in mind (new issue?), currently all labels on the Preference panel are OK now,  I think (and also note [this suggestion](https://github.com/textpattern/textpattern/issues/1840#issuecomment-1851793314) on adding `aria-describedby` for the `yes/no` radio buttons).

Closing
